Src/CSharpier/SyntaxPrinter/TrailingComma.cs

```diff
@@ -0,0 +1,16 @@
+namespace CSharpier.SyntaxPrinter;
+
+internal static class TrailingComma
+{
+    public static Doc Print(SyntaxToken nextSyntaxToken, FormattingContext context)
+    {
+        return
+            context.UsePrettierStyleTrailingCommas
+            && !nextSyntaxToken.LeadingTrivia.Any(o => o.IsDirective)
```

Comment:
Is there a test for this? I don't quite follow why the trailing comma is only added when there is not a directive.

Reply:
Added a test for this behavior.
And yeah, this one is kinda lazy, because I don't really know how to make sure we don't add a comma if there is a leading comma in the directive, e.g.:
`public enum EnumDeclarationWithLeadingCommaInDirectiveBeforeBracket { 
    Foo = 1
#if DEBUG
    ,
    Bar = 2
#endif
}`

Reply:
ah yeah, conditional compilation causes a few headaches like this